hom/utils.py

```diff
@@ -48,63 +53,73 @@ def get_channel_by_id(
-    return message_mentions[0] if message_mentions else None
-
+async def get_user_by_original_message(
+    channel: discord.TextChannel,
+) -> t.Optional[discord.Member | discord.User]:
+    messages = [message async for message in channel.history(limit=1, oldest_first=True)]
+    mentions = messages[0].mentions if messages else None
+    return mentions[0] if mentions else None
 
-def get_user_ticket_channel(guild, user):
-    category = get_category(
-        discord_guild=guild, channel_name="Tickets"
-    )  # TODO: Make this channel_id?
 
-    if not category:
-        return
+def get_user_ticket_channel(
+    guild: discord.Guild, user: discord.User | discord.Member
+) -> t.Optional[discord.guild.GuildChannel]:
+    if category := get_category(guild, "Tickets"):  # TODO: Make this channel_id?
+        for channel in category.channels:
+            user_perms = channel.overwrites_for(user)
 
-    for channel in category.channels:
-        user_perms = channel.overwrites_for(user)
-        if user_perms.view_channel:
-            return channel
+            if user_perms.view_channel:
+                return channel
 
     return None
 
 
-async def create_ticket_for_user(interaction, instructions, button_label, example_url=None):
-    prevent_dupe_channels = True
+async def create_ticket_for_user(
+    interaction: discord.Interaction,
+    instructions: str,
+    button_label: t.Optional[str],
+    example_url: t.Optional[str] = None,
+) -> discord.TextChannel:
+    assert interaction.guild
+    assert isinstance(interaction.user, discord.Member)
     existing_ticket_channel = get_user_ticket_channel(interaction.guild, interaction.user)
 
-    if existing_ticket_channel and prevent_dupe_channels:
+    if existing_ticket_channel:
         msg_content = f":envelope:  Click [here]({existing_ticket_channel.jump_url}) to view your open ticket."
         await interaction.followup.send(content=msg_content, ephemeral=True)
-        return
+        return t.cast(discord.TextChannel, existing_ticket_channel)
 
-    tickets_category = get_category(
-        discord_guild=interaction.guild, channel_name="Tickets"
-    )  # TODO: Make this channel_id?
     channel_name = f"help-{interaction.user.display_name[:15]}"
-
+    tickets_category = get_category(interaction.guild, "Tickets")  # TODO: Make this channel_id?
     mod_role = get_role_by_name(interaction.guild, "Moderator")
     helpful_old_man_role = get_role_by_name(interaction.guild, "Helpful Old Man")
-    new_text_channel = await interaction.channel.guild.create_text_channel(
+    assert helpful_old_man_role
+    assert mod_role
+
+    new_text_channel = await interaction.guild.create_text_channel(
         name=channel_name,
         category=tickets_category,
         reason=f"{interaction.user.display_name} ({interaction.user.id}) has opened a ticket.",
-        topic=button_label,
+        topic=button_label or "Unknown (This is a bug)",
```

Comment:
Something tells me this isn't entirely accurate. If we provide an empty string for the button_label being provided (not sure if we would/should/do) then that's not necessarily "Unknown (This is a bug)", it would just be a blank channel topic.

I think the more correct thing to do would be to require a topic parameter when using the provided method.